src/openrct2-ui/windows/ServerList.cpp

```diff
@@ -496,7 +504,7 @@ static void server_list_save_server_entries()
 {
     // Get total number of favourite servers
     sint32 count = 0;
-    for (sint32 i = 0; i < _numServerEntries; i++) {
+    for (sint32 i = 0; i < _serverEntries.size(); i++) {
```

Comment:
This could also be done with a foreach loop, but I'm not sure how far you want to take this PR.